lib/processors/index.js

```diff
@@ -4,50 +4,71 @@
  * Dependencies
  */
 
-var parseContentType = require('content-type').parse;
 var parseHtml = require('magnet-html-parser');
-var ua = require('../../config').user_agent;
 var debug = require('debug')('processor');
-var request = require('superagent');
+var request = require('../utils/request');
 
 /**
  * Exports
  */
 
-module.exports = function(url) {
+module.exports = function exec(url, options) {
   debug('processing', url);
-  return get(url)
+  var adaptors = options && options.adaptors;
+
+  // requests can optionally come with 'adaptor'
+  // urls, which is an alternative endpoint
+  var adaptorUrl = findAdaptorUrl(adaptors, url);
+  if (adaptorUrl) return exec(adaptorUrl);
+
+  return request.get(url)
     .then(res => {
-      var type = getType(res);
-      if (type !== 'text/html') throw new Error('unsupported type: ' + type);
 
-      // final url (after redirects)
+      // final-url (after redirects)
       var endUrl = res.request.url;
+      debug('got response', endUrl);
+
+      // requests can optionally come with 'adaptor'
+      // urls, which is an alternative endpoint
+      var adaptorUrl = findAdaptorUrl(adaptors, endUrl);
+      if (adaptorUrl) return exec(adaptorUrl);
+
+      var contentType = res.headers['content-type'];
+
+      // if the service returned json, we don't need to parse anything
+      if (isType(contentType, 'application/json')) {
+        debug('url returned json');
+        return res.body;
+      }
+
+      // only handle html content from here on
+      if (!isType(contentType, 'text/html')) {
+        debug('unsupported content-type', contentType);
+        throw new Error('unsupported type: ' + contentType);
```

Comment:
I don't think we should throw error, this will cause the server to die. console.error perhaps?